lib/avrora/storage/registry.ex

```diff
@@ -33,7 +33,7 @@ defmodule Avrora.Storage.Registry do
          {:ok, id} <- Map.fetch(response, "id"),
          {:ok, version} <- Map.fetch(response, "version"),
          {:ok, schema} <- Map.fetch(response, "schema"),
-         {:ok, schema} <- Schema.parse(schema) do
+         {:ok, schema} <- Schema.parse(schema, csr_references_lookup_function(response)) do
```

Comment:
Since this function is local and will be called on each reference we should apply a few adjustments:

1. Rename `csr_references_lookup_function` to `make_references_lookup_fun` because the scope of the whole module is CSR so no need to mention it in any name here
2. All the references **must** be prepared in advance otherwise (like in the current implementation) they will be extracted on each function call

As an idea:

```suggestion
         {:ok, references} <- extract_references(response),
         {:ok, schema} <- Schema.parse(schema, make_references_lookup_fun(references)) do
```

where references is a `Map`